vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/FolderArtifactHandler.java

```diff
@@ -62,6 +70,18 @@ public void setNodeType(String nodeType) {
         this.nodeType = nodeType;
     }
 
+    private Node createIntermediateNode(Node parent, String intermediateNodeName) throws RepositoryException {
+        // preferably use default (=primary) node type for intermediate nodes
+        NodeType primaryNodeType = parent.getPrimaryNodeType();
+        final Node node;
+        if (!DISALLOWED_PRIMARY_NODE_TYPE_NAMES.contains(primaryNodeType.getName()) && 
```

Comment:
@kwin  - this doesn't solve the issue at hand - it is not the primaryType of the parent that is the problem but the default nodetype - therefore the default nodetype would need to be detected and compared instead (IIUC thats part of the NodeDefinitions that can be obtained via getChildNodeDefinitions())